schemas/typescript/types.ts

```diff
@@ -217,30 +217,117 @@ export interface IntentRequest {
 }
 
 /**
- * Status enum for intent submission responses.
+ * Status enum for order submission responses.
  * @description Distinguishes between successful receipt and validation failures at the discovery stage.
  * Note: Full validation (oracle routes, etc.) happens asynchronously after receipt.
  */
-export enum IntentResponseStatus {
-  /** Intent received and passed basic validation, queued for full validation */
+export enum PostOrderResponseStatus {
+  /** Order received and passed basic validation, queued for full validation */
   Received = "received",
-  /** Intent rejected due to validation failure */
+  /** Order rejected due to validation failure */
   Rejected = "rejected",
-  /** Intent processing encountered an error */
+  /** Order processing encountered an error */
   Error = "error",
 }
 
 /**
- * Response from intent submission
- * @description Response from intent submission
+ * Response from order submission
+ * @description Response from order submission
  */
-export interface IntentResponse {
+export interface PostOrderResponse {
   /** Assigned order identifier if accepted */
   orderId?: string;
-  /** Status of the intent submission */
-  status: IntentResponseStatus;
+  /** Status of the order submission */
+  status: PostOrderResponseStatus;
   /** Optional message for additional details on status */
   message?: string;
   /** The submitted EIP-712 typed data order */
   order?: Record<string, unknown>;
 }
+
+/**
+ * Order status enum for order tracking
+ * @description Tracks the lifecycle of an order from submission through execution
+ */
+export enum OrderStatus {
+  /** Order has been created but not yet prepared */
+  Created = "created",
+  /** Order is pending execution */
+  Pending = "pending",
+  /** Order has been executed */
+  Executed = "executed",
+  /** Order has been settled and is ready to be claimed */
+  Settled = "settled",
+  /** Order is finalized and complete (after claim confirmation) */
+  Finalized = "finalized",
+  /** Order execution failed */
+  Failed = "failed",
+}
+
+/**
+ * Asset amount representation using ERC-7930 interoperable address format
+ * @description Asset amount representation using ERC-7930 interoperable address format
+ */
+export interface AssetAmount {
+  /** Asset address in ERC-7930 interoperable format */
+  asset: Address;
+  /** Amount as a string-encoded integer */
+  amount: Amount;
+}
+
+/**
+ * Settlement mechanism types
+ * @description Settlement mechanism types
+ * TODO: Add lock types to this schema
+ */
+export enum SettlementType {
+  /** Escrow-based settlement */
+  Escrow = "escrow",
+  /** Resource lock-based settlement */
+  ResourceLock = "resourceLock",
+}
```

Comment:
@pepebndc not sure if we can re-use your `originSubmission` structure here as well for auth schemes?

We currently just return this in Solvers, but we need to be more elaborate IMO.

Reply:
What I was thinking was extracting "schemes" into a separate type and then reusing both here and in originSubmission?

Reply:
I'll manage that in a later PR if-and-when we make the change to solvers as well.